docs/using.md

```diff
@@ -23,10 +25,14 @@ When you create the consumer you specify the consumer group as follows:
 topic_consumer = client.get_topic_consumer(os.environ["input"], consumer_group = "empty-transformation")
 ```
 
-!!! note
+Best practice is to make sure the consumer group name matches the name of the service.
+
+!!! warning
 
     If you don't specify a consumer group, then all messages in all streams in a topic will be processed by all replicas in the microservice deployment.
```

Comment:
No, not if you've not specified a consumer group, because in your case both streams will be processed by both replicas. This is basically why consumer groups exist, to allow you to specify horizontal scaling. As was pointed out in the conversation linked below, for some use cases you do want all replicas to process all streams.

My understanding is based on [this conversation](https://quixworld.slack.com/archives/C04KXLCGQ94/p1678805872511799). If you don't specify a consumer group, then all replicas will process all streams.

Reply:
OK, I had been understanding this wrong for some time then. I'll need to do some code refactoring :S